Tharun/Assignment_2.cu

```diff
@@ -7,7 +7,7 @@ __global__ void Sum(float* d_in1,float* d_in2, float* d_out,int* d_array_size_x,
 	int j = threadIdx.x + blockIdx.x * blockDim.x;
     int k = threadIdx.y + blockIdx.y * blockDim.y;
     
-	int i = j + k * *d_array_size_y;
+	int i = k + j * *d_array_size_y;
 	
     if (k < *d_array_size_y && j < *d_array_size_x) 
```

Comment:
This if condition should be (j < *d_array_size_y && k < *d_array_size_x).
This is because you have defined j as the index in the x dimension (horizontal direction) and the maximum index limit in the x dimension is the number of columns of the matrix, which you have define as d_array_size_y. And similarly the maximum limit for k (defined as the index in y dimension) is the number of rows in the matrix, which has been defined as d_array_size_x.

Reply:
Thank you! Got it